Debounce/Generic/Debouncer.cs

```diff
@@ -5,452 +5,116 @@
+                }
+                _DataLimit = value;
+                if (TriggerData.Count >= _DataLimit)
+                {
+                    LockedSendEvent();
+                }
             }
         }
-        else
-        {
-            // We were already disposed.
-            Interlocked.Exchange(ref InterlockedCountMinusOne, long.MinValue);
-            throw new ObjectDisposedException(GetType().FullName);
-        }
     }
 
-    /// <inheritdoc/>
-    public long Reset()
+    /// <inheritdoc />
+    public void Trigger(TData data)
     {
         lock (LockObject)
         {
-            lock (TriggerDataLock)
+            if (IsDisposed)
             {
-                TriggerData = [];
+                throw new ObjectDisposedException(GetType().FullName);
             }
-            if (!IsDisposed)
+            if (TriggerData.Count >= DataLimit)
             {
-                Count = AddWithClamp(Count, Interlocked.Exchange(ref InterlockedCountMinusOne, -1) + 1);
+                throw new InvalidOperationException("Trigger data reached limit while already executing event handler.");
             }
-            var count = Count;
-            Count = 0;
-            if (!IsDisposed)
+            // Must add data first, as Trigger() may decide to invoke the Debounced event including the current trigger.
+            TriggerData.Add(data);
+            Trigger();
+            // Additional event reason
+            if (!SendingEvent && (TriggerData.Count >= _DataLimit))
             {
-                LockedReschedule();
+                LockedSendEvent();
             }
-            return count;
         }
     }
 
-    TimeSpan GetField(ref TimeSpan field)
+    IReadOnlyList<TData> LockedExchangeTriggerData()
     {
-        lock (LockObject)
+        if (TriggerData.Count == 0)
         {
-            return field;
+            // Prevent unnecessary reallocations.
+            return EmptyTriggerData;
         }
-    }
-
-    void SetField(ref TimeSpan field, TimeSpan value, bool allowInfinite, [CallerMemberName] string fieldName = "")
-    {
-        lock (LockObject)
+        else
         {
-            ThrowIfDisposed();
-            if (field == value)
-            {
-                return;
-            }
-            if (value == Timeout.InfiniteTimeSpan)
-            {
-                if (!allowInfinite)
-                {
-                    throw new ArgumentOutOfRangeException(fieldName, $"{fieldName} must not be infinite (-1 ms).");
-                }
-            }
-            else if (value < TimeSpan.Zero)
-            {
-                throw new ArgumentOutOfRangeException(fieldName, $"{fieldName} must not be negative.");
-            }
-            field = value;
-            LockedReschedule();
+            return new ReadOnlyCollection<TData>(Interlocked.Exchange(ref TriggerData, []));
```

Comment:
Nice usage of the `Interlocked.Exchange()`! although you're already locked so no specific need for an atomic operation, right?